backend/app/services/metrics_service.py

```diff
@@ -36,13 +37,40 @@ class LatencyMeasurement(NamedTuple):
 class MetricsService:
     """Process-safe metrics service using shared memory for model metric data"""
 
+    _shared_memory: ClassVar[SharedMemory | None] = None
+    _instance_count: ClassVar[int] = 0
+    _lock_class: ClassVar[LockType] = Lock()
```

Comment:
This design will not work reliably in a multiprocess setting because each process gets its own separate instance of `MetricsService`, including separate instances of the lock and shared memory variables. For safe and correct IPC, primitives like shared memory and locks should be instantiated once in a central place (e.g., the `Scheduler`) and then explicitly passed down to downstream processes. This ensures all processes coordinate on the same shared resources and synchronization objects, preventing race conditions and data inconsistencies.